cmd/boskos/boskos.go

```diff
@@ -144,3 +172,69 @@ func main() {
 	// signal to the world that we're ready
 	health.ServeReady()
 }
+
+type configSyncReconciler struct {
+	sync func() error
+}
+
+func (r *configSyncReconciler) Reconcile(_ reconcile.Request) (reconcile.Result, error) {
+	err := r.sync()
+	if err != nil {
+		logrus.WithError(err).Error("Config sync failed")
+	}
+	return reconcile.Result{}, err
+}
+
+func addConfigSyncReconcilerToManager(mgr manager.Manager, configSync func() error, configChangeEvent <-chan event.GenericEvent) error {
+	ctrl, err := controller.New("bokos_config_reconciler", mgr, controller.Options{
+		// We reconcile the whole config, hence this is not safe to run concurrently
+		MaxConcurrentReconciles: 1,
+		Reconciler: &configSyncReconciler{
+			sync: configSync,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	if err := ctrl.Watch(&source.Kind{Type: &crds.ResourceObject{}}, constHandler(), resourceUpdatePredicate()); err != nil {
+		return fmt.Errorf("failed to watch boskos resources: %w", err)
+	}
+	if err := ctrl.Watch(&source.Kind{Type: &crds.DRLCObject{}}, constHandler()); err != nil {
+		return fmt.Errorf("failed to watch boskos dyanmic resources: %w", err)
+	}
+	if err := ctrl.Watch(&source.Channel{Source: configChangeEvent}, constHandler()); err != nil {
+		return fmt.Errorf("failed to create source channel for config change event: %w", err)
+	}
+	if err := mgr.Add(ctrl); err != nil {
+		return fmt.Errorf("failed to add controller to manager: %w", err)
+	}
+
+	return nil
+}
+
+func constHandler() handler.EventHandler {
+	return &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(handler.MapObject) []reconcile.Request {
+			return []reconcile.Request{{}}
+		}),
+	}
+}
+
+// resourceUpdatePredicate prevents the config reconciler from reacting to resource update events
+// except if the new status is Tombstone, because in that case we have to delete it.
+func resourceUpdatePredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool { return true },
+		DeleteFunc: func(_ event.DeleteEvent) bool { return true },
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			resource, ok := e.ObjectNew.(*crds.ResourceObject)
+			if !ok {
+				panic(fmt.Sprintf("BUG: expected *crds.ResourceObject, got %T", e.ObjectNew))
+			}
+
+			return resource.Status.State == common.Tombstone
```

Comment:
This is needed because boskos is the entity that deletes Tombstoned resources: https://github.com/kubernetes-sigs/boskos/blob/eea29b1a07663e131ea8eb5c8f8b6537e36ad8ee/ranch/storage.go#L291

The process is:
* Boskos sets to ToBeDeleted
* Cleaner cleans and sets to Tombstone
* Boskos deletes

To me it seems that the cleaner should just delete them. The only additional thing we do here is to check that they have no owner, but they will always have no owner because the cleaner always does both clean, then release: https://github.com/kubernetes-sigs/boskos/blob/eea29b1a07663e131ea8eb5c8f8b6537e36ad8ee/cleaner/cleaner.go#L88-L90

Are there more reasons or can we (in a follow-up) make the cleaner just delete the resources?

Reply:
cc @sebastienvas 

This seems reasonable to me. There may have been some reason this was separated that got removed in subsequent refactorings.